src/base/Memory.cu

```diff
@@ -14,8 +14,6 @@
 
 namespace celeritas
 {
-namespace detail
-{
 //---------------------------------------------------------------------------//
 void device_memset(void* data, int fill_value, size_type count)
```

Comment:
We are accumulating a fair number of helper functions that are mostly undocumented (ie. uncommented as to their purpose).  In this case, its use/purpose is pretty straightforward, but this could fairly quickly become overwhelming.  I think we want at least a small (`\brief`) doxygen block that gives the purpose here, and then perhaps, and ultimately, a directory of service functions in the developer documentation?  

I don't want to make the same "mistakes" we made in Exnihilo where we have a slew of service functions that are largely known through word-of-mouth as opposed to more formal software engineering practice.

Reply:
Good call; I think that leaving them "undocumented" is borderline OK if they're in the detail namespace, but moving them into the  main code should definitely require new documentation.  I'll add it. I've also set up an issue to generate doxygen: https://github.com/celeritas-project/celeritas/issues/27